server/sonar-server/src/main/java/org/sonar/server/component/ws/SuggestionsAction.java

```diff
@@ -112,33 +114,41 @@ SuggestionsWsResponse doHandle(String query) {
       return Optional.empty();
     }
 
-    List<Component> results = fetchFromDatabase(uuids)
+    List<ComponentDto> componentDtos;
+    Map<String, String> organizationKeyByUuids;
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      componentDtos = dbClient.componentDao().selectByUuids(dbSession, uuids);
+      organizationKeyByUuids = getOrganizationKeys(dbSession, componentDtos);
+    }
+
+    List<Component> results = componentDtos
       .stream()
-      .map(SuggestionsAction::dtoToComponent)
+      .map(dto -> dtoToComponent(dto, organizationKeyByUuids))
       .collect(Collectors.toList());
 
     Qualifier q = Qualifier.newBuilder()
       .setQ(qualifier)
       .addAllItems(results)
       .build();
+
     return Optional.of(q);
   }
 
-  private List<String> searchInIndex(ComponentIndexQuery componentIndexQuery) {
-    return index.search(componentIndexQuery);
+  private Map<String, String> getOrganizationKeys(DbSession dbSession, List<ComponentDto> componentDtos) {
+    return dbClient.organizationDao().selectByUuids(
+      dbSession,
+      componentDtos.stream().map(ComponentDto::getOrganizationUuid).collect(Collectors.toSet()))
+      .stream()
+      .collect(Collectors.uniqueIndex(OrganizationDto::getUuid, OrganizationDto::getKey));
   }
 
-  private List<ComponentDto> fetchFromDatabase(List<String> uuids) {
-    DbSession dbSession = dbClient.openSession(false);
-    try {
-      return dbClient.componentDao().selectByUuids(dbSession, uuids);
-    } finally {
-      dbClient.closeSession(dbSession);
-    }
+  private List<String> searchInIndex(ComponentIndexQuery componentIndexQuery) {
+    return index.search(componentIndexQuery);
   }
 
-  private static Component dtoToComponent(ComponentDto result) {
+  private static Component dtoToComponent(ComponentDto result, Map<String, String> organizationKeysByUuid) {
     return Component.newBuilder()
+      .setOrganization(organizationKeysByUuid.get(result.getOrganizationUuid()))
```

Comment:
If the organization does not exist, then request will fail with a NullPointerException. Not a pb so far as we don't expect orphans, but still a clear message would be great.